services/QuillLMS/engines/evidence/spec/controllers/evidence/research/gen_ai/llm_examples_controller_spec.rb

```diff
@@ -0,0 +1,33 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+module Evidence
+  module Research
+    module GenAI
+      RSpec.describe LLMExamplesController, type: :controller do
+        routes { Evidence::Engine.routes }
+
+        let(:dataset) { create(:evidence_research_gen_ai_dataset) }
+        let(:trial) { create(:evidence_research_gen_ai_trial, dataset_id: dataset.id) }
+        let!(:llm_example) { create(:evidence_research_gen_ai_llm_example, trial_id: trial.id) }
+        let(:user) { double('User') }
+        let(:parsed_response) { JSON.parse(response.body) }
+
+        before do
+          allow(controller).to receive(:current_user).and_return(user)
+          allow(user).to receive(:staff?).and_return(true)
+        end
+
+        describe 'GET #index' do
+          it 'returns all llm examples for the trial' do
+            get :index, params: { dataset_id: dataset.id, trial_id: trial.id }, format: :json
+
+            expect(response).to have_http_status(:ok)
+            expect(parsed_response['llm_examples']).to eq(JSON.parse(trial.llm_examples.to_json))
```

Comment:
Can we use `trial.llm_examples.as_json` instead of `JSON.parse(trial.llm.examples.to_json)`?  I suspect that `as_json` adds some additional keys that we wouldn't expect to be in the raw JSON, but that might make this a good candidate for `include` over `eq`.  `expect(A).to include(B)` will pass if A includes at least all values in B, but A is allowed to have additional data beyond that.  If there is any data in B that is not in A, then the test will fail.

This test may actually be specific enough that inclusion isn't detailed enough, in which case I'd say leave it as is.